cmd/statusd/main.go

```diff
@@ -106,6 +108,12 @@ func main() {
 		go startCollectingStats(interruptCh, backend.NodeManager())
 	}
 
+	// Sync blockchain and stop.
+	if *exitWhenSynced {
+		syncStopErr := backend.NodeManager().SyncAndStopNode(time.Minute * 50)
+		defer handleSyncStopErr(syncStopErr)
```

Comment:
I am not really sure why `defer` was chosen. I would suggest making `SyncAndStopNode` synchronous and actually, change this method a bit (more about that below).

```go
err := backend.NodeManager().SyncAndStopNode(*exitWhenSynced * time.Minute)
if err != nil {
  log.Fatalf("Failed to sync: %v", err)
}
return
```

Reply:
I just wanted it to play along with the existing flow and make it such as:
1) Start node.
2) Launch sync asynchronously.
3) Wait until node is stopped.
4) Handle aggregated errors from sync (max 2).

So I used `defer` there to handle aggregated errors before exiting the program and it makes the least amount of modification to main function thus more readable.

Reply:
I think I missed the fact that reused code just tracks the progress, sorry about that. :)